app/src/main/java/org/mozilla/focus/browser/integration/FullScreenIntegration.kt

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.browser.integration
+
+import android.app.Activity
+import android.os.Build
+import android.view.View
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.feature.session.FullScreenFeature
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.feature.UserInteractionHandler
+import org.mozilla.focus.browser.DisplayToolbar
+
+class FullScreenIntegration(
+    val activity: Activity,
+    val sessionManager: SessionManager,
+    tabId: String?,
+    private val toolbarView: DisplayToolbar,
+    private val engineView: EngineView,
+    private val statusBar: View
+) : LifecycleAwareFeature, UserInteractionHandler {
+    private val feature = FullScreenFeature(
+        sessionManager,
+        SessionUseCases(sessionManager),
+        tabId,
+        ::viewportFitChanged,
+        ::fullScreenChanged
+    )
+
+    override fun start() {
+        feature.start()
+    }
+
+    override fun stop() {
+        feature.stop()
+    }
+
+    private fun fullScreenChanged(enabled: Boolean) {
+        if (enabled) {
+            toolbarView.visibility = View.GONE
+
+            toolbarView.setExpanded(false, true)
+            statusBar.visibility = View.GONE
+
+            engineView.setDynamicToolbarMaxHeight(0)
```

Comment:
We should remove this line (and in the else-branch) as well since it only helps when we have a toolbar at the bottom of the screen. Otherwise, it causes weird web content bugs with bottom toolbars like this one on youtube:

<img width="634" alt="Screen Shot 2020-06-22 at 10 28 22 PM" src="https://user-images.githubusercontent.com/1370580/85354094-c720b800-b4d7-11ea-8623-c7c01dad9cdc.png">